contrib/mkimage/debootstrap

```diff
@@ -97,26 +97,49 @@ if [ -d "$rootfsDir/etc/apt/apt.conf.d" ]; then
-	# In Docker, we don't often need the "Translations" files, so we're just wasting
-	# time and space by downloading them, and this inhibits that.  For users that do
-	# need them, it's a simple matter to delete this file and "apt-get update". :)
+		# In Docker, we don't often need the "Translations" files, so we're just wasting
+		# time and space by downloading them, and this inhibits that.  For users that do
+		# need them, it's a simple matter to delete this file and "apt-get update". :)
 
-	Acquire::Languages "none";
+		Acquire::Languages "none";
 	EOF
 
 	echo >&2 "+ echo Acquire::GzipIndexes 'true' > '$rootfsDir/etc/apt/apt.conf.d/docker-gzip-indexes'"
 	cat > "$rootfsDir/etc/apt/apt.conf.d/docker-gzip-indexes" <<-'EOF'
-	# Since Docker users using "RUN apt-get update && apt-get install -y ..." in
-	# their Dockerfiles don't go delete the lists files afterwards, we want them to
-	# be as small as possible on-disk, so we explicitly request "gz" versions and
-	# tell Apt to keep them gzipped on-disk.
+		# Since Docker users using "RUN apt-get update && apt-get install -y ..." in
+		# their Dockerfiles don't go delete the lists files afterwards, we want them to
+		# be as small as possible on-disk, so we explicitly request "gz" versions and
+		# tell Apt to keep them gzipped on-disk.
 
-	# For comparison, an "apt-get update" layer without this on a pristine
-	# "debian:wheezy" base image was "29.88 MB", where with this it was only
-	# "8.273 MB".
+		# For comparison, an "apt-get update" layer without this on a pristine
+		# "debian:wheezy" base image was "29.88 MB", where with this it was only
+		# "8.273 MB".
 
-	Acquire::GzipIndexes "true";
-	Acquire::CompressionTypes::Order:: "gz";
+		Acquire::GzipIndexes "true";
+		Acquire::CompressionTypes::Order:: "gz";
+	EOF
+
+	# update "autoremove" configuration to be aggressive about removing suggests and recommends deps that weren't manually installed
+	echo >&2 "+ echo Apt::AutoRemove::RecommendsImportant 'false' > '$rootfsDir/etc/apt/apt.conf.d/docker-autoremove-recommends'"
+	cat > "$rootfsDir/etc/apt/apt.conf.d/docker-autoremove-recommends" <<-'EOF'
+		# Since Docker users are looking for the smallest possible final images, the
+		# following emerges as a very common pattern:
+
+		#   RUN apt-get update \
+		#       && apt-get install -y <packages> \
+		#       && <do some compilation work> \
+		#       && apt-get purge -y --auto-remove <packages>
+
+		# By default, APT will actually _keep_ packages installed via Recommends if
+		# another package even Suggests them, even and including if the package that
+		# originally caused them to be installed is removed.  Setting these to "false"
+		# ensures that APT is appropriately aggressive about removing the packages it
+		# added.
+
+		# https://aptitude.alioth.debian.org/doc/en/ch02s05s05.html#configApt-AutoRemove-RecommendsImportant
+		Apt::AutoRemove::RecommendsImportant "false";
```

Comment:
In re-reading the descriptions of these, I'm thinking maybe only `Apt::AutoRemove::SuggestsImportant` makes sense by default, because `Recommends` is actually a sane dependency, where `Suggests` is a much more "loose" dependency.

Ideally, we'd be able to have these just match `Install-Recommends` and `Install-Suggests`, but that's not really possible here that I'm aware of.

cc @paultag (can I trouble you for further thoughts on this topic?)